download-directory/Jenkinsfile

```diff
@@ -0,0 +1,65 @@
+import groovy.json.JsonSlurper
+
+pipeline {
+    	agent any
+
+	stages {
+		stage('Check upstream builds succeeded'){
+			steps{
+				script{
+					def addLinksInsertionStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/AddLinks-Insertion/lastBuild/api/json"
+					def addLinksInsertionJson = new JsonSlurper().parseText(addLinksInsertionUrl.getContent())
+					if(addLinksInsertionJson['result'] != "SUCCESS"){
+						error("Most recent AddLinks-Insertion build status: " + addLinksInsertionJson['result'] + ". Please complete a successful build.")
+					}
+			    	}
+			}
+		}
+	    
+		stage('Setup: Install Pathway Exchange artifact'){
+			steps{
+				script{
+					dir('download-directory'){
+						sh "./build_pathway_exchange.sh"
+					}
+				}
+			}
+		}
+        	stage('Setup: Build DownloadDirectory archive'){
+            		steps{
+				script{
+                    			dir('download-directory'){
+						sh "mvn clean package -DskipTests"
+			    			sh "rm -rf downloadDirectory"
+						sh "unzip -o target/downloadDirectory-distr.zip"
```

Comment:
What is `downloadDirectory-distr.zip`, and why does it get unzipped?

Reply:
This is how download-directory gets packaged. Since it depends on a lot of files stored in `src/main/resources`, it basically packages all of them, with the downloadDirectory jar into an archive. This could be one of those things that I've done in an elaborate and unnecessary way, but this is how download directory has always worked.

Reply:
Right. Because the BioPax step needs to run with that "javaagent" argument, and that requires everything to be bundled up in a zip file.